regolith/helpers/a_todohelper.py

```diff
@@ -0,0 +1,157 @@
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+
+        rc.coll = f"{TARGET_COLL}"
+        rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        rc = self.rc
+        if not rc.id:
+            try:
+                rc.id = rc.default_user_id
+            except AttributeError:
+                print(
+                    "Please set default_user_id in '~/.config/regolith/user.json',\
+                     or you need to enter your group id in the command line")
+                return
+        try:
+            person = document_by_value(all_docs_from_collection(rc.client, "people"), "_id", rc.id)["name"]
+        except TypeError:
+            print(
+                "The id you entered can't be found in people.yml.")
+            return
+
+        now = dt.date.today()
+        if not rc.begin_date:
+            begin_date = now
+        else:
+            begin_date = date_parser.parse(rc.begin_date).date()
+
+        try:
+            # if the entered rc.due_date is an integer:
+            relative_day = int(rc.due_date)
+            due_date = begin_date + relativedelta(days=relative_day)
+        except ValueError:
+            due_date = date_parser.parse(rc.due_date).date()
+
+        if begin_date > due_date:
+            raise ValueError("begin_date can not be after due_date")
+
+        importance = int(rc.importance)
+        if importance not in ALLOWED_IMPORTANCE:
+            raise ValueError(f"importance should be chosen from {ALLOWED_IMPORTANCE}")
+
+        person = rc.client.find_one(rc.database, rc.coll, {'_id': rc.id})
+        if person:
+            # if this person exists in the database
+            if person.get("todos"):
+                # if this person has key "todos", just append this new task to todos:
+                person["todos"].append({
+                    'description': rc.description,
+                    'due_date': due_date,
+                    'begin_date': begin_date,
+                    'duration': float(rc.duration),
+                    'importance': importance,
+                    'status': "started"})
+            else:
+                # if this person doesn't has key "todos", "todos" will be created
+                person.update({"todos": [{
+                    'description': rc.description,
+                    'due_date': due_date,
+                    'begin_date': begin_date,
+                    'duration': float(rc.duration),
+                    'importance': importance,
+                    'status': "started"}]
+                })
+                if rc.notes:
```

Comment:
wouldn't it be cleaner to update the dict before you appended/update the dict.  Doing it this way is a bit ugly.